File: UVA/10226_Happywood_Species/solution.py
cases = int(input())
_ = input()

for i in range(cases):
    species = {}
    while 1:
        try:
            tree_name = input()
            if tree_name == "":
                break
            elif tree_name in species:
                species[tree_name] += 1
            else:
                species[tree_name] = 1
        except:
            break

    total_species = sum(species.values())

    for key in sorted(species.keys()):
        print(f"{key} {species[key] / total_species * 100:.4f}")

    if i != cases - 1:
        print()

File: UVA/10226_Happywood_Species/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    long long cases;
    cin >> cases;
    string line;
    getline(cin, line);
    getline(cin, line);

    for (long long i = 0; i < cases; i++) {
        map<string, double> species;
        string tree_name;
        double total_species = 0;
        while (1) {
            getline(cin, tree_name);
            if (tree_name != "") {
                species[tree_name]++;
                total_species++;
            } else {
                break;
            }
        }

        for (auto& it: species) {
            cout << fixed << setprecision(4);
            cout << it.first << " " << it.second * (double)100 / total_species << endl;
        }

        if (i != cases - 1) {
            cout << endl;
        }
    }

    return 0;
}